lib/protocol/protocol.js

```diff
@@ -31,8 +31,65 @@ const IMG_EL_URL_RE = new RegExp(
 
 class Protocol {}
 
-function getLogByProtocol (driver) {
-  return driver.isW3CProtocol() ? w3cLog : mjsonwpLog;
+function getLogByProtocol (protocol) {
+  switch (protocol) {
+    case BaseDriver.DRIVER_PROTOCOL.W3C:
+      return w3cLog;
+    case BaseDriver.DRIVER_PROTOCOL.MJSONWP:
+      return mjsonwpLog;
+    default:
+      return genericProtocolLog;
+  }
+}
+
+// This cache is useful when a session gets terminated
+// and removed from the sessions list in the umbrella driver,
+// but the client still tries to send a command to the same session id.
+// So we know how to properly wrap the error message for it
+const PROTOCOLS_MAPPING_CACHE = {};
```

Comment:
we already use an lru cache module for other purposes, perhaps it could simplify this?

Reply:
I've almost forgotten about it (considering it was my own suggestion :) ) 
Updated now.

Reply:
ha!